includes/admin/class-wc-admin-notices.php

```diff
@@ -488,6 +492,20 @@ public static function redirect_download_method_notice() {
 		include dirname( __FILE__ ) . '/views/html-notice-redirect-only-download.php';
 	}
 
+	/**
+	 * Notice about uploads directory begin public.
+	 *
+	 * @since 4.2.0
+	 */
+	public static function uploads_directory_is_public_notice() {
+		if ( ! self::is_uploads_directory_is_public() || get_user_meta( get_current_user_id(), 'dismissed_uploads_directory_is_public_notice', true ) ) {
```

Comment:
if we swap the conditions, then the (potentially expensive) check does not need to happen if the user dismissed the notice, no?